aasemble/django/apps/mirrorsvc/models.py

```diff
@@ -71,6 +77,7 @@ def write_config(self):
 
     @property
     def basepath(self):
+        ensure_dir(os.path.join(settings.MIRRORSVC_BASE_PATH, 'mirrors'))
         d = os.path.join(settings.MIRRORSVC_BASE_PATH, 'mirrors', str(self.uuid))
         if not os.path.isdir(d):
             os.makedirs(d)
```

Comment:
I'm not sure why this is necessary. os.makedirs should create all the missing directories.